src/main/java/com/google/devtools/build/remote/client/GrpcRemoteCache.java

```diff
@@ -66,16 +70,33 @@ public static boolean isRemoteCacheOptions(RemoteOptions options) {
     return options.remoteCache != null;
   }
 
+  private static ClientInterceptor customHeadersInterceptor(Map<String, String> headers) {
+    Metadata metadata = new Metadata();
+    for (Map.Entry<String,String> entry : headers.entrySet()) {
+      metadata.put(
+        Metadata.Key.of(entry.getKey(), Metadata.ASCII_STRING_MARSHALLER),
+        entry.getValue()
+      );
+    }
```

Comment:
Two things spring to mind here: encodings, and the repetition of header key entries (multiple of the same key) - I was pretty sure that grpc/http2 headers matched with http1, where repetitions of keys was accepted.

Might want to try that with some non-ascii compatible UTF8 or some such, to see the result.

Reply:
Yeah, repetition should append. I'll work on both.

Reply:
Got encoding, bazel handles it way for this flag as well, so I think we are good there: https://github.com/bazelbuild/bazel/blob/4de6bcaa2d0489f91746c5700fbe99f49fa9d0c5/src/main/java/com/google/devtools/build/lib/remote/util/TracingMetadataUtils.java#L110-L124

Reply:
It seems it handles repetition with the same code as well.

Reply:
@werkt